.github/workflows/unit-tests.yml

```diff
@@ -11,7 +11,7 @@ jobs:
   run:
     strategy:
       matrix:
-        platform: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
Are you sure about this? We can make the step that does `sudo make test` conditional and only include `sudo` on macOS and Linux. Wouldn't that be better?

Reply:
The thing is. The unit tests dont run for Windows on `main` either, it just fails silently instead. But when splitting it out in its own steps it fails with a make error. Since I do not have a windows machine I rather not get into debugging the Cmake+CTest on Windows...

I would say it sufficient to run the unit tests under Linux and mac